File: 45-jump-game-ii/jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int jumps = 0;
        int l = 0;
        int r = 0;
        int n = nums.length;
        while (r < n-1){
            int farthest = 0;
            for(int index = l; index <= r; index++){
                farthest = Math.max(farthest, index + nums[index]);
            }
            l = r + 1;
            r = farthest;
            jumps++;
        }

        return jumps;
    }
}

File: 45-jump-game-ii/jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        farthest = 0
        curr_end = 0
        n = len(nums)
        jump = 0

        for i in range(n-1):
            farthest  = max(farthest, i+nums[i])

            if i == curr_end:
                jump+=1
                curr_end = farthest

        return jump